File: backend/app/schemas/learning.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional

class LearningRecordCreate(BaseModel):
    poem_id: int
    action: str
    duration: int = 0

class UserStatsResponse(BaseModel):
    total_learned: int
    total_favorites: int
    study_time: int
    streak_days: int
    level: int
    exp: int
    next_level_exp: int

class FavoriteResponse(BaseModel):
    id: int
    poem_id: int
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/learning.ts
import request from '@/utils/request'

export interface UserStats {
  total_learned: number
  total_favorites: number
  study_time: number
  streak_days: number
  level: number
  exp: number
  next_level_exp: number
}

export interface LearningProgress {
  daily_study_time: Array<{ date: string; duration: number }>
  dynasty_distribution: Array<{ dynasty: string; count: number }>
  genre_distribution: Array<{ genre: string; count: number }>
  cumulative_learned: Array<{ date: string; total: number }>
  challenge_performance: {
    beauty_avg: number
    creativity_avg: number
    mood_avg: number
  }
  study_calendar: Array<{ date: string; activity: number }>
}

export const learningApi = {
  recordLearning(data: {
    poem_id: number
    action: string
    duration?: number
  }) {
    return request.post('/learning/record', data)
  },

  getStats() {
    return request.get<UserStats>('/learning/stats')
  },

  getProgress() {
    return request.get<LearningProgress>('/learning/progress')
  }
}
